File: src/main/java/com/IconPippi/simtriggers/wrappers/Menu.java
package com.IconPippi.simtriggers.wrappers;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import com.IconPippi.simtriggers.events.EventFactory;
import com.IconPippi.simtriggers.utils.Logger;

import flightsim.simconnect.SimConnect;

/**
 * This class simplifies the menu creating process
 * @author IconPippi
 */
public class Menu {
	
	/* SimConnect constant */
	private final SimConnect sc = SimTriggers.getSimulator();
	
	/* Logger */
	private final Logger logger = new Logger();
	
	/*
	 * Menu data
	 */
	private String menuName;
	private String title;
	private List<String> options = new ArrayList<>();
	private String menuHandler;
	private int menuID;
	
	/**
	 * Initialize a menu object by providing its handler
	 * @param menuHandler Handler's function name
	 */
	public Menu(String menuHandler) {
		this.menuHandler = menuHandler;
		
		menuID = new EventFactory().buildMenu(this.menuHandler);
	}
	
	/**
	 * Sets the menu's name (a.k.a "title" in the SimConnect .menu method)
	 * @param name Menu's name
	 */
	public void setName(String name) {
		this.menuName = name;
	}
	
	/**
	 * Sets the menu's title (a.k.a "prompt" in the SimConnect .menu method)
	 * @param title Menu's title
	 */
	public void setTitle(String title) {
		this.title = title;
	}
	
	/**
	 * Add a option to the menu and its action specifying the function's name
	 * @param option Option name
	 */
	public void addOption(String option) {
		options.add(option);
	}
	
	/**
	 * Show the menu
	 */
	public void show() {
		try {
			sc.menu(0, menuID, menuName, title, options.toArray(new String[0]));
		} catch (UnsupportedOperationException | IOException e) {
			logger.error(e.toString());
		}
	}
	
	/**
	 * Hide the menu
	 */
	public void hide() {
		try {
			sc.menu(0.0f, menuID, null, null, (String[])null); //Hide the menu
		} catch (UnsupportedOperationException | IOException e) {
			logger.error(e.toString());
		}
	}
}

File: src/main/kotlin/com/simtriggers/fsx/simulator/Menu.kt
package com.simtriggers.fsx.simulator

import com.simtriggers.fsx.SimTriggers
import com.simtriggers.fsx.event.EventRegisterer
import flightsim.simconnect.SimConnect
import java.util.ArrayList

/**
 * 06/08/2019
 * This class represents a FSX menu
 *
 * @author IconPippi
 */
class Menu(private val menuHandler: String) {

    /** Simulator constant */
    private val sc: SimConnect = SimTriggers.sc

    /** Event registerer */
    private val eventRegisterer: EventRegisterer = EventRegisterer()

    /** Menu identifier */
    private val menuID: Int

    private var name: String? = null
    private var title: String? = null
    private var options = ArrayList<String>()

    /**
     * Register a new Menu instance
     */
    init {
        menuID = eventRegisterer.registerMenuEvent(menuHandler)
    }

    /**
     * Set the menu's name
     * @param name menu's name
     */
    fun setName(name: String) {
        this.name = name
    }

    /**
     * Set the menu's title
     * @param title menu's title
     */
    fun setTitle(title: String) {
        this.title = title
    }

    /**
     * Add a menu option
     * @param option Option text
     */
    fun addOption(option: String) {
        options.add(option)
    }

    /**
     * Show the menu
     */
    fun show() {
        sc.menu(0f, menuID, name, title, *options.toArray((arrayOfNulls<String>(0))))
    }

    /**
     * Hide the menu
     */
    fun hide() {
        sc.menu(0.0f, menuID, "", "", "")
    }

}